tests/test_query.py

```diff
@@ -319,3 +273,161 @@ async def mock_get(
         "Subjects cannot both be healthy controls and have a diagnosis"
         in response.text
     )
+
+
+@pytest.mark.parametrize("valid_min_num_sessions", [1, 2, 4, 7])
+def test_get_valid_sensible_min_num_sessions(
+    test_app, mock_successful_get, valid_min_num_sessions, monkeypatch
+):
+    """Given a valid and sensible minimum number of imaging sessions, returns a 200 status code and a non-empty list of results."""
+
+    monkeypatch.setattr(crud, "get", mock_successful_get)
+    response = test_app.get(
+        f"/query/?min_num_sessions={valid_min_num_sessions}"
+    )
+    assert response.status_code == 200
+    assert response.json() != []
+
+
+def test_get_valid_nonsensical_min_num_sessions(test_app, monkeypatch):
+    """Given a valid and snonsensical minimum number of imaging sessions, returns a 200 status code and an empty list of results."""
+
+    async def mock_get(
+        age_min,
+        age_max,
+        sex,
+        diagnosis,
+        is_control,
+        min_num_sessions,
+        image_modal,
+    ):
+        return []
+
+    monkeypatch.setattr(crud, "get", mock_get)
+    response = test_app.get("/query/?min_num_sessions=1000")
+    response.status_code = 200
+    response.json() == []
+
+
+@pytest.mark.parametrize("invalid_min_num_sessions", [0, -3, "apple"])
+def test_get_invalid_min_num_sessions(
+    test_app, invalid_min_num_sessions, monkeypatch
+):
+    """Given an invalid minimum number of imaging sessions, returns a 422 status code."""
+
+    async def mock_get(
```

Comment:
sorry for bringing this all up here, but I don't think we need to even mock the get method for invalid parameters. If the internal validation does its job, then there shouldn't be any get request going out that we would have to mock. So we could just call the API without the mock and expect it to fail. That would also make the test more legible, becuase we're not asserting over the returned result list being empty (because that doesn't make sense here), so it's not immediately clear why we are mocking the getter to return an empty list.

Reply:
Because you have already reshuffled some of the tests here, I think it would be fine for you to also get rid of the "unhappy-path-test-get-mock" stuff.

Reply:
No thanks for bringing it up, I've had this question in my head for a while but somehow forgot to look into it or even ask about it.
I can go through the tests and do a refactor.